src/nupic/engine/RegionImpl.cpp

```diff
@@ -230,7 +230,11 @@ void RegionImpl::setParameterArray(const std::string& name, Int64 index,const Ar
       break;
     }
 
-    NTA_ASSERT(rc == 0) << "getParameterArray - failure to get parameter '" << name << "' on node of type " << getType();
+    if (rc != 0)
```

Comment:
Does this really have the same semantics? I would have guessed that `NTA_THROW` would have been the equivalent of `NTA_CHECK`, not `NTA_ASSERT`.

But regardless, I prefer the `NTA_ASSERT`/`NTA_CHECK` version since it is more compact.